### PR TITLE
Merge editor now compare incoming changes with common ancestor commit

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The list of changes in the merge editor is long and not easy to comprehend.

Issue Number: #388 

### What is the new behavior?

Instead of comparing the incoming with the current model, the incoming model is compared with a common ancestor node. As a result, the change set only contains the changes made in the incoming branch. Those changes can be applied to the model.

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: gaphor/ui/modelmerge/tests/test_organize_uml.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def test_applied_stereotype(element_factory, modeling_language, create):
     apply_stereotype(class_item.subject, stereotype)
 
     current = ElementFactory()
-    all(compare(current, element_factory))
+    all(compare(current, current, element_factory))
```
Comment (danyeaw) [inline gaphor/ui/modelmerge/tests/test_organize_uml.py:185]: For my understanding, are these tests now comparing the current against itself?

Comment (amolenaar) [reply gaphor/ui/modelmerge/tests/test_organize_uml.py:185]: They conpare current against element_factory. The resulting changes are created in the current model.

I have to admit, the api is still a bit messy.

Comment (danyeaw) [reply gaphor/ui/modelmerge/tests/test_organize_uml.py:185]: Thanks for the explanation!

**Review by danyeaw — APPROVED** [review]
Nice update :+1:

**Comment by amolenaar** [comment]
Hi @bayerl,

This behavior is activated when you have a merge conflict in git. So if you make changes on your branch, then open the model on the main branch and make some changes. If you merge the branch into main, and there are merge conflicts, you can still open the model woth Gaphor. This merge editor will show up and allow you to selectively add changes from your branch to the main branch.

**Comment by bayerl** [comment]
@amolenaar thanks for the fast response! Ok I got it. This approach makes sense. Very cool feature! I will be experimenting with this further.

I did see the docs, which helped me discover this feature in the first place. I've suggested a slight modification that would've made it clear to me in PR #2914

**Comment by amolenaar** [comment]
@bayerl Thanks. I merge the PR.

**Pull request merged**